core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -393,7 +394,9 @@ public boolean deleteRoute(Route route) {
 	 * @return Hostname as is expected to be generated by OpenShift
 	 */
 	public String generateHostname(String routeName) {
-		if(routeSuffix == null) {
+		if (StringUtils.isNotBlank(cz.xtf.core.config.OpenShiftConfig.routeDomain())) {
+			routeSuffix = cz.xtf.core.config.OpenShiftConfig.routeDomain();
+		} else if (routeSuffix == null) {
```

Comment:
Could you please move this check out off this `if() {} else if () {}` block? This way you are checking every time you call the method and do the reassignment, but it's rather lazy initialization and any subsequent call will get you same result.